tests/basics/builtin_range_slice.py

```diff
@@ -0,0 +1,16 @@
+# test builtin range type slicing difficult cases
+# These should pass if
+#define MICROPY_PY_BUILTINS_SLICE_RANGES_CORRECTLY (1)
+
+try:
+    range(4)[::0]
+except ValueError:
+    print("ValueError")
+else:
+    print("SKIP")
+    import sys
+    sys.exit(0)
+
+print(range(7,-2,-4)[2:-2:])
```

Comment:
Note that, even though uPy gives the wrong range for this, it does give an empty range which is correct.  Ie, `list(range(7,-2,-4)[2:-2:])` is correct under uPy.